core/src/main/java/org/apache/gravitino/catalog/ManagedTableOperations.java

```diff
@@ -0,0 +1,555 @@
+import org.apache.gravitino.rel.TableCatalog;
+import org.apache.gravitino.rel.TableChange;
+import org.apache.gravitino.rel.expressions.Expression;
+import org.apache.gravitino.rel.expressions.distributions.Distribution;
+import org.apache.gravitino.rel.expressions.sorts.SortOrder;
+import org.apache.gravitino.rel.expressions.transforms.Transform;
+import org.apache.gravitino.rel.indexes.Index;
+import org.apache.gravitino.rel.indexes.Indexes;
+import org.apache.gravitino.rel.types.Type;
+import org.apache.gravitino.storage.IdGenerator;
+import org.apache.gravitino.utils.PrincipalUtils;
+
+public abstract class ManagedTableOperations implements TableCatalog {
+
+  private static final Joiner DOT = Joiner.on(".");
+
+  protected abstract EntityStore store();
+
+  protected abstract SupportsSchemas schemas();
+
+  protected abstract IdGenerator idGenerator();
+
+  @Override
+  public NameIdentifier[] listTables(Namespace namespace) throws NoSuchSchemaException {
+    try {
+      NameIdentifier schemaIdent = NameIdentifier.of(namespace.levels());
+      schemas().loadSchema(schemaIdent); // verify schema existence
+
+      List<TableEntity> tables =
+          store().list(namespace, TableEntity.class, Entity.EntityType.TABLE);
+      return tables.stream()
+          .map(t -> NameIdentifier.of(namespace, t.name()))
+          .toArray(NameIdentifier[]::new);
+
+    } catch (NoSuchEntityException e) {
+      throw new NoSuchSchemaException(e, "Schema %s does not exist", namespace);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to list tables in schema " + namespace, e);
+    }
+  }
+
+  @Override
+  public Table loadTable(NameIdentifier ident) throws NoSuchTableException {
+    try {
+      TableEntity tableEntity = store().get(ident, Entity.EntityType.TABLE, TableEntity.class);
+      return toGenericTable(tableEntity);
+
+    } catch (NoSuchEntityException e) {
+      throw new NoSuchTableException(e, "Table %s does not exist", ident);
+    } catch (IOException e) {
+      throw new RuntimeException("Failed to load table " + ident, e);
+    }
+  }
+
+  @Override
+  public Table createTable(
+      NameIdentifier ident,
+      Column[] columns,
+      String comment,
+      Map<String, String> properties,
+      Transform[] partitions,
+      Distribution distribution,
+      SortOrder[] sortOrders,
+      Index[] indexes)
+      throws NoSuchSchemaException, TableAlreadyExistsException {
+    // createTable in ManagedTableOperations only stores the table metadata in the entity store.
+    // It doesn't handle any additional operations like creating physical location, preprocessing
+    // the properties, etc. Those operations should be handled in the specific catalog
+    // implementation.
+
+    NameIdentifier schemaIdent = NameIdentifier.of(ident.namespace().levels());
+
+    if (!schemas().schemaExists(schemaIdent)) {
+      throw new NoSuchSchemaException("Schema %s does not exist", schemaIdent);
+    }
```

Comment:
It seems unnecessary, since we can catch `NoSuchEntityException` when call `store().put(tableEntity, false /* overwrite */)`